spmd/tensor/utils.py

```diff
@@ -31,20 +31,20 @@ def wrap(res: object, spec: OutputSpecType) -> object:
     if isinstance(res, torch.Tensor):
         assert spec is not None and isinstance(
             spec, DTensorSpec
-        ), "output spec does not match with output!"
+        ), f"output spec does not match with output! Expected DTensorSpec, got {spec}."
         return spmd_tensor.DTensor(res, spec.mesh, spec.placements)
     elif isinstance(res, list):
         assert spec is not None and isinstance(
             spec, list
-        ), "output spec does not match with output!"
+        ), f"output spec does not match with output! Expected list, got {spec}."
         return list(
             spmd_tensor.DTensor(e, s.mesh, s.placements)
             for e, s in zip(res, spec)
         )
     elif isinstance(res, tuple):
         assert spec is not None and isinstance(
             spec, tuple
-        ), "output spec does not match with output!"
+        ), f"output spec does not match with output! Expected tuple, got {spec}"
```

Comment:
nit: I would inline {spec} next to output spec and add "tuple" in brackets next to output.